### ISSUE TITLE
Invalid syntax freezes the JsonPreprocessor

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#381]
In the following JSONP snippet, parts of the import path are extracted from a list and a dictionary:

```
"dictDirs" : {"AA" : "AA", "BB" : "BB", "CC" : "CC", "DD" : "DD"},
"listDirs" : ["AA", "BB", "CC", "DD"],
"[import]" : "./dynamic_imports/${dictDirs}['AA']/${listDirs}[1]/${dictDirs.CC}/${listDirs.3}/imported.${listDirs.3}.1.jsonp"
```

This runs properly.

The following two versions of the third line contain syntax errors at different positions (closing square bracket instead of closing curly bracket after '`listDirs.3`'):

```
"[import]" : "./dynamic_imports/${dictDirs}['AA']/${listDirs}[1]/${dictDirs.CC}/${listDirs.3}/imported.${listDirs.3].1.jsonp"
"[import]" : "./dynamic_imports/${dictDirs}['AA']/${listDirs}[1]/${dictDirs.CC}/${listDirs.3]/imported.${listDirs.3}.1.jsonp"
```

In both cases the bracket mismatch is detected properly:

`Error: 'Invalid syntax! One or more than one closed curly bracket is missing in expression ...`

The following import contains the bracket mismatch two times:

`"[import]" : "./dynamic_imports/${dictDirs}['AA']/${listDirs}[1]/${dictDirs.CC}/${listDirs.3]/imported.${listDirs.3].1.jsonp"`

Result: JsonPreprocessor freezes.

Expected: bracket mismatch error message

**Comment by HolQue** [comment]
Shorter expression to reproduce the freeze:

`"[import]" : "./${${AA/imported.jsonp"`

**Comment by HolQue** [comment]
Seems to be a top-level problem. The bracket mismatch is detected properly (without freeze) if this occurs within
an imported file.

**Comment by test-fullautomation** [comment]
due to possible freezing also prio1

**Comment by HolQue** [comment]
JPP_1157

**Comment by namsonx** [comment]
Hello Holger,

This ticket was solved on stabi branch, could you please verify it again.

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
merged for version 0.13.1
